### PR TITLE
MH-13710: Fix a bug in paella loader plugin when a track has no tags

### PR DESCRIPTION
**Pull request by miesgre**
Jira ticket: https://opencast.jira.com/browse/MH-13710

**Steps to reproduce:**
1. Process a mediapackage with a track without tags
2. Try to play with paella

**Actual Results:**
Paella fails to load the video

**Expected Results:**
Paella should play the video

### CODE REVIEW

File: modules/engage-paella-player/src/main/paella-opencast/plugins/es.upv.paella.opencast.loader/03_oc_search_converter.js [hunk 1]
```diff
@@ -215,11 +215,17 @@ class OpencastToPaellaConverter {
...
+      let importT = false;
+      if (currentTrack.tags.tag) {
+        if (!(currentTrack.tags.tag instanceof Array)) {
+          currentTrack.tags.tag = [currentTrack.tags.tag];
+        }
+        importT = filterStream.tracks.tags.some(function(cTag) {
+          return currentTrack.tags.tag.some(function(t){
+            return ((cTag == '*') || (cTag == t));
+          });
         });
```
Comment (karendolan) [inline modules/engage-paella-player/src/main/paella-opencast/plugins/es.upv.paella.opencast.loader/03_oc_search_converter.js]: @miesgre This solves the problem is an Opencast mediapackage TRACK element has no tags such as to "archive" the track or to publish it to "engage".... if the track was never tagged to be published to the search index. Where is Paella getting it from? I suspect this is related to the other recent "bug fix"  issue where error handing is needed for published events which have no creation date nor even title.

Comment (lkiesow) [reply modules/engage-paella-player/src/main/paella-opencast/plugins/es.upv.paella.opencast.loader/03_oc_search_converter.js]: I think, I see where the confusion comes from: @karendolan, you don't need to tag a track to publish to engage. The `publish-engage` operation has always supported publication by flavor only. Tagging tracks is just a convenient way to mark things for publication without having to remember later on what needs publication.

Comment (miesgre) [reply modules/engage-paella-player/src/main/paella-opencast/plugins/es.upv.paella.opencast.loader/03_oc_search_converter.js]: @karendolan Paella is getting the information from the search service. As @lkiesow there is no need to tag a track to upload to the engage. See the `download-source-flavors` key of the `publish-engage` operation.  But I found this problem on live events. When there is a live events, the tracks haven't tags.

**Review by lkiesow — COMMENTED** [review]
Looking at the code, it seems that if `cTag == '*'`, tracks without tags would not be selected. Is that intentional? My expectation for configuring a global wildcard (`*`) would be that no filter applies. But I've also not looked into this very deeply.

**Comment by miesgre** [comment]
> Looking at the code, it seems that if `cTag == '*'`, tracks without tags would not be selected. Is that intentional? My expectation for configuring a global wildcard (`*`) would be that no filter applies. But I've also not looked into this very deeply.

It's not intentional. I'm going to fix it.

**Comment by miesgre** [comment]
@lkiesow Fixed. You can continue to review it

**Comment by lkiesow** [comment]
There still seems to be a problem.
I get:

```
paella_player.js:4380 Uncaught (in promise) TypeError: Cannot read property 'currentTime' of null
    at paella_player.js:4380
```

I used the following workflow:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<definition xmlns="http://workflow.opencastproject.org">

  <id>import</id>
  <title>Import Mediapackage and Publish without reencode</title>
  <tags>
  </tags>
  <description>
  A minimal Workflow that only publishes a Mediapackage
  </description>

  <operations>

    <!-- Ingest elements from external working file repositories and file servers -->

    <operation
      id="ingest-download"
      description="Ingesting external elements">
      <configurations>
        <configuration key="delete-external">true</configuration>
      </configurations>
    </operation>

    <!-- Apply ACL from series to the mediapackage -->

    <operation
      id="series"
      fail-on-error="true"
      exception-handler-workflow="error"
      description="Applying access control entries from series">
      <configurations>
        <configuration key="apply-acl">true</configuration>
      </configurations>
    </operation>

    <!-- Inspect the media -->

    <operation
      id="inspect"
      fail-on-error="true"
      exception-handler-workflow="error"
      description="Inspecting audio and video streams">
      <configurations>
        <configuration key="overwrite">false</configuration>
        <configuration key="accept-no-media">false</configuration>
      </configurations>
    </operation>

    <!-- Publish to engage player -->

    <operation
      id="publish-engage"
      max-attempts="2"
      fail-on-error="true"
      exception-handler-workflow="error"
      description="Publishing to Engage">
      <configurations>
        <configuration key="download-source-flavors">dublincore/*,security/*,*/delivery</configuration>
        <configuration key="check-availability">true</configuration>
      </configurations>
    </operation>

    <!-- Archive the current state of the mediapackage -->

    <operation
      id="snapshot"
      fail-on-error="true"
      exception-handler-workflow="error"
      description="Archiving">
      <configurations>
        <configuration key="source-tags">archive</configuration>
      </configurations>
    </operation>

    <!-- Clean up the working file repository -->

    <operation
      id="cleanup"
      fail-on-error="false"
      description="Cleaning up">
      <configurations>
        <configuration key="delete-external">true</configuration>
      </configurations>
    </operation>

  </operations>

  <state-mappings>
    <state-mapping state="running">EVENTS.EVENTS.STATE_MAPPING.IMPORTING</state-mapping>
    <state-mapping state="failing">EVENTS.EVENTS.STATE_MAPPING.IMPORTING</state-mapping>
  </state-mappings>

</definition>
```

and ingested a video using:

```sh
curl -f -i --digest -u opencast_system_account:CHANGE_ME -H "X-Requested-Auth: Digest" http://localhost:8080/ingest/addMediaPackage/import -F 'flavor=presentation/delivery' -F 'BODY=@test.mp4' -F title="`uuidgen`" -F creator="`uuidgen`"
```

**Comment by miesgre** [comment]
@lkiesow Fixed. You can continue the review

**Comment by lkiesow** [comment]
Works. Merged.

**Pull request merged**